common/engine/keyboardprocessor/src/kmx/kmx_options.cpp

```diff
@@ -18,7 +18,7 @@ void KMX_Options::AddOptionsStoresFromXString(PKMX_WCHAR s) {
       case CODE_SAVEOPT:
       case CODE_RESETOPT:
         idx = *(s + 2) - 1;
-        if (idx >= 0 && idx < _kp->Keyboard->cxStoreArray && _kp->Keyboard->dpStoreArray[idx].dpName != NULL) {
+        if (idx >= 0 && idx < (int) _kp->Keyboard->cxStoreArray && _kp->Keyboard->dpStoreArray[idx].dpName != NULL) {
```

Comment:
it shouldn't happen that cxStoreArray is too large but maybe cast `(KMX_DWORD) idx` instead since it is already known to be >= 0 ?

Reply:
Also we should be using function style or explicit casts rather than C-style